### PR TITLE
Fix rad2deg and deg2rad precision for integer and float64 inputs

### LINKED ISSUE #3937: rad2deg/deg2rad: float32 kornia.constants.pi is cast to the input dtype — integer inputs divide by 3, float64 loses ~7 digits
`kornia.geometry.conversions.rad2deg` and `deg2rad` divide by `kornia.constants.pi` after casting that constant to the **input** tensor's dtype (`conversions.py:97` and `:127`: `180.0 * tensor / pi.to(tensor.device).type(tensor.dtype)`). `pi` is created as `torch.tensor(3.14159265358979323846)` (`kornia/constants.py:25`), which materialises as a **float32** tensor. Two consequences follow: for an **integer** input tensor the constant is cast to int64 and becomes literally `3`, so the conversion is off by ~5 %; for a **float64** input the constant is upcast from its already-truncated float32 value, so ~7 significant digits are silently lost. `angle_to_rotation_matrix` calls `deg2rad` and inherits the integer bug directly. `torch.rad2deg` / `torch.deg2rad` are correct in both cases.

## Repro

```python
import math
import torch
from kornia.constants import pi
from kornia.geometry.conversions import rad2deg, deg2rad, angle_to_rotation_matrix

print("pi.dtype, pi.item():", pi.dtype, pi.item())
print("pi.type(torch.int64):", pi.type(torch.int64))
print("pi.type(torch.float64).item():", pi.type(torch.float64).item(), "| math.pi:", math.pi)

# (1) integer input
print("kornia rad2deg int:", rad2deg(torch.tensor([1, 2, 3])))
print("torch  rad2deg int:", torch.rad2deg(torch.tensor([1, 2, 3])))
print("kornia deg2rad int:", deg2rad(torch.tensor([180, 90])))
print("torch  deg2rad int:", torch.deg2rad(torch.tensor([180, 90])))

# (2) downstream: angle_to_rotation_matrix with an integer angle
print("angle_to_rotation_matrix(tensor([90])) :", angle_to_rotation_matrix(torch.tensor([90])))
print("angle_to_rotation_matrix(tensor([90.])):", angle_to_rotation_matrix(torch.tensor([90.])))

# (3) float64 precision
x64 = torch.tensor(math.pi, dtype=torch.float64)
print("kornia rad2deg f64 err:", rad2deg(x64).item() - 180.0)
print("torch  rad2deg f64 err:", torch.rad2deg(x64).item() - 180.0)
d64 = torch.tensor(180.0, dtype=torch.float64)
print("kornia deg2rad f64:", deg2rad(d64).item(), "| torch:", torch.deg2rad(d64).item(), "| math.pi:", math.pi)
print("kornia deg2rad f64 abs err:", abs(deg2rad(d64).item() - math.pi))
```

## Observed

```
pi.dtype, pi.item(): torch.float32 3.1415927410125732
pi.type(torch.int64): tensor(3)
pi.type(torch.float64).item(): 3.1415927410125732 | math.pi: 3.141592653589793
kornia rad2deg int: tensor([ 60., 120., 180.])
torch  rad2deg int: tensor([ 57.2958, 114.5916, 171.8873])
kornia deg2rad int: tensor([3.0000, 1.5000])
torch  deg2rad int: tensor([3.1416, 1.5708])
angle_to_rotation_matrix(tensor([90])) : tensor([[[ 0.0707,  0.9975],
         [-0.9975,  0.0707]]])
angle_to_rotation_matrix(tensor([90.])): tensor([[[-4.3711e-08,  1.0000e+00],
         [-1.0000e+00, -4.3711e-08]]])
kornia rad2deg f64 err: -5.008956179608504e-06
torch  rad2deg f64 err: 0.0
kornia deg2rad f64: 3.1415927410125732 | torch: 3.141592653589793 | math.pi: 3.141592653589793
kornia deg2rad f64 abs err: 8.742278012618954e-08
```

## Expected

- `rad2deg(torch.tensor([1, 2, 3]))` should be `[57.2958, 114.5916, 171.8873]`, matching `torch.rad2deg`. Observed `[60., 120., 180.]` is exactly what you get from dividing by 3 instead of π — the output dtype is already promoted to float32, so nothing about the integer input forces an integer constant.
- `angle_to_rotation_matrix(torch.tensor([90]))` should equal `angle_to_rotation_matrix(torch.tensor([90.]))` (the quarter-turn matrix). Observed is the matrix for `90 * 3 / 180 = 1.5` rad ≈ 85.94°.
- `deg2rad(torch.tensor(180., dtype=torch.float64))` should be `3.141592653589793` (what `torch.deg2rad` returns). Observed `3.1415927410125732` carries an absolute error of `8.74e-08` — about **7 significant digits of the 16 that float64 provides**. Symmetrically `rad2deg(π, f64)` is off 180° by `-5.0e-06` where `torch.rad2deg` is exact.

Note that `rad2deg(deg2rad(x))` still round-trips cleanly, because the same biased constant appears in the numerator and denominator and cancels — which is why this has gone unnoticed.

Suggested fix: delegate to `torch.rad2deg` / `torch.deg2rad`, or at minimum build the constant at the input's dtype from a float64 source rather than casting a float32 tensor.

To size the blast radius on the common float32 path, here is the exact comparison snippet (fixed seed, four input ranges spanning six orders of magnitude — absolute deltas are scale-dependent and not meaningful here, so this reports ULP distance and relative error):

```python
import torch
from kornia.geometry.conversions import rad2deg, deg2rad

def cmp(name, fn_k, fn_t, x):
    a, b = fn_k(x), fn_t(x)
    same = (a == b).float().mean().item()
    rel = (a - b).abs() / b.abs().clamp(min=torch.finfo(torch.float32).tiny)
    ulp = (a.view(torch.int32) - b.view(torch.int32)).abs().max().item()
    print(f"{name:>24}: bit-identical {same * 100:5.1f}%  max rel {rel.max().item():.3e}  max ULP {ulp}")

g = torch.Generator().manual_seed(0)
ranges = {
    "U(-10,10)": torch.rand(100000, generator=g) * 20 - 10,
    "U(-1e3,1e3)": torch.rand(100000, generator=g) * 2e3 - 1e3,
    "U(-1e6,1e6)": torch.rand(100000, generator=g) * 2e6 - 1e6,
    "U(0,1)": torch.rand(100000, generator=g),
}
for n, x in ranges.items():
    cmp("rad2deg " + n, rad2deg, torch.rad2deg, x)
for n, x in ranges.items():
    cmp("deg2rad " + n, deg2rad, torch.deg2rad, x)
```

```
       rad2deg U(-10,10): bit-identical  52.2%  max rel 1.869e-07  max ULP 2
     rad2deg U(-1e3,1e3): bit-identical  52.1%  max rel 1.869e-07  max ULP 2
     rad2deg U(-1e6,1e6): bit-identical  52.2%  max rel 1.867e-07  max ULP 2
          rad2deg U(0,1): bit-identical  51.9%  max rel 1.864e-07  max ULP 2
       deg2rad U(-10,10): bit-identical  57.1%  max rel 1.675e-07  max ULP 2
     deg2rad U(-1e3,1e3): bit-identical  55.6%  max rel 1.673e-07  max ULP 2
     deg2rad U(-1e6,1e6): bit-identical  55.2%  max rel 1.676e-07  max ULP 2
          deg2rad U(0,1): bit-identical  55.9%  max rel 1.676e-07  max ULP 2
```

So on float32 roughly half the values stay bit-identical and the rest move by **at most 2 ULP** (max relative error ~1.9e-07, i.e. about one float32 epsilon) — stable across all four ranges. Meanwhile the integer and float64 paths go from wrong to correct.

## Classification

**Clearly-broken-output bug**, not a semantic/default-convention change. The integer path returns numbers that are wrong by ~5 % under any reading of "convert radians to degrees" — there is no convention under which π equals 3 — and the float64 path is a plain precision defect against an exact reference. Per the reference-core lifecycle (Part I), this class may land early as an ordinary bugfix rather than waiting for the coordinated Repair & Deprecation window. The ≤2 ULP float32 movement measured above is the only behavior change on the common path.

## Environment

kornia 0.9.0rc1 @ `3358ef16`, torch 2.9.1, cpu.

Posted on behalf of @ducha-aiki by Claude (Fable 5).

### PR DESCRIPTION
**Pull request by Hrithikcrick**
Fixes #3937

## Summary

- Fix `rad2deg` and `deg2rad` for integer tensor inputs.
- Preserve float64 precision using Python's high-precision `math.pi`.
- Use basic tensor arithmetic instead of `torch.rad2deg` / `torch.deg2rad` to preserve ONNX export compatibility.
- Update the existing #3937 regression tests.
- Remove obsolete warnings related to the fixed behavior.
- Update `kornia.feature.siftdesc` to import `pi` directly from `kornia.constants`.
- Update half-precision known-failure lists because the affected tests now pass.

## Testing

- `pytest tests/geometry/test_conversions.py -k "3937" -q`
  - 6 passed

- `pytest tests/geometry/test_conversions.py -q`
  - 393 passed, 1 skipped, 8 xfailed

- `pytest tests/augmentation/test_onnx_export.py -k "RandomAffine or RandomRotation" -q`
  - 5 passed

- `pixi run -e default lint`
  - ruff check passed
  - ruff format passed

- GitHub Actions
  - 39 checks passed

### CODE REVIEW

**Review by ducha-aiki — COMMENTED** [review]
The conversion implementation and its regression coverage look correct, and this is the preferable version of the overlapping #4356 fix. Two documentation edits remain before merge: update the comments at `tests/geometry/test_conversions.py:2554-2571`, which still say all three operations are broken and marked strict xfail even though this patch removes that mark and fixes the outputs; and add the required `CHANGELOG.md` bug-fix entry citing #4358.

Validation at `eabe5de5f0d96efd68c12a5d9c287d84dd5a244b`: `import kornia` resolves to the PR worktree; 620 CPU float32/float64 conversion tests passed (2 skipped, 12 expected xfails); all 39 hosted checks are green. `pi` was never in `conversions.__all__` or the API-surface inventory, and the in-tree SIFT consumer now imports it directly from `kornia.constants`. No prior comments or formal reviews exist.

Posted on behalf of [@ducha-aiki](https://github.com/ducha-aiki) by Codex (gpt-5.6-sol).

**Comment by Hrithikcrick** [comment]
Addressed the requested follow-up changes: updated the stale #3937 regression comments and added the CHANGELOG.md bug-fix entry for #4358. Verified with the #3937 tests and lint; all pass.

**Review by ducha-aiki — DISMISSED** [review]
Delta re-review at `8576fec8`.

Both requested follow-ups are resolved: the #3937 regression comments now describe the repaired behavior, and `CHANGELOG.md` has the required Bug fixes entry citing #4358. The implementation remains the previously reviewed angle-conversion fix; the merge from upstream places it on current `main` without conflicts.

Validation at this exact head: 620 CPU float32/float64 conversion tests passed (2 skipped, 12 expected xfails), `git diff --check` is clean, all 40 hosted checks are green, and `import kornia` resolved to the isolated PR worktree.

Posted on behalf of [@ducha-aiki](https://github.com/ducha-aiki) by Codex (gpt-5.6-sol).
